Cargo.toml

```diff
@@ -25,6 +25,7 @@ name = "crypto"
 [features]
 default = [ ]
 std = [ ]
+aes-cbc = [ "aes-crate", "block-modes", "cipher", "hmac", "sha", "subtle" ]
```

Comment:
AES-related features or doc-comment in `src/ciphers/traits.rs` for `Aead` trait seem a bit broken. The following commands fail (but should not):

```
cargo test --features="aes-cbc"
cargo test --features="aes-cbc aes"
```

Reply:
Specifying required features for cargo tests is a pain point, so I think one typically runs `cargo test --all-features`.

I think the alternative for these cases is

> cargo test --features="aes-cbc"

Make `aes-cbc` part of the `aes` feature completely, and don't expose it as a separate feature at all,

> cargo test --features="aes-cbc aes"

Activate the `random` feature whenever the `aead` feature is enabled, but that is bad style, imo, because we would only do that for tests.

I think the current solution is okay, and tests should be run with all features enabled.

Reply:
I think using `--all-features` is a bad idea as it hides problems with bad `#[cfg(feature...)]` attributes for imports or definitions which become unavailable if a certain feature is not enabled. This should be resolved in `Cargo.toml` with the right feature dependencies and with the right cfg attributes, and not by user enabling all features.

The failing doc-test requires `aes` feature which might not be enabled. Also, `aes` feature implies aes gcm, but not other modes (cbc, keywrap), which is also misleading.

I'd propose:
1. rename `aes` feature to `aes-gcm`;
2. create `aes = ["aes-cbc", "aes-gcm", "aes-kw"]` feature to enable all aes-related modes;
3. move doc-test in ciphers::traits into aes-gcm tests, or create a dedicated example, or somehow introduce `#[cfg(feature = "aes-gcm")]` to the doc-test.

Reply:
I did the renaming as you suggested. In the process, I also renamed the `aes` module to `aes_gcm` so it is in line with the other modules `aes_cbc` and `aes_kw`. Note that this is a breaking change and requires major version bump.

I feature gated the test you mentioned. It always succeeds if the required features are not enabled. No way around that, I think. The two test invocations you posted succeed now. Note that CI will still run the test because it uses `--all-features`, so it's fine overall.

Please have a look at the definition of the `aes` feature and if you agree with that.